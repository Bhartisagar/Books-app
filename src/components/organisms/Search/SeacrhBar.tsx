import { makeStyles, TextField, Typography,Grid,Container ,InputAdornment} from '@material-ui/core';
import Cards from '../../molecules/BookCard';
import { useEffect,useState } from 'react';
import { Search } from '@material-ui/icons';
import { useNavigate } from 'react-router';
import Banner from '../../molecules/Banner/Banner';

const useStyles= makeStyles({
  
   mainDiv:{
     marginTop:'100px',    
      
   },
   typo:{
    marginLeft:'17%',
    marginTop:'20px' ,
    fontWeight:700,
   
    width:'40%'
   },
   search:{
    marginLeft:'17%',
    marginTop:'20px' ,
    fontWeight:700,
    color:'grey',
    width:'40%',
    fontSize:'24px',
    
    
   },
   
  
  

 })
 interface DataBook{
   
  "id": number,
  "imgSrc": string,
  "imgAlt": string,
  "name": string,
  "author": string,
  "readTime": string,
  "readCount": string,
  "state": string,
  "category": string,
  "audio": boolean,
  "synopsis"?:string,
"justAdded": boolean,
"isTrending": boolean

}

const SeacrhBar=()=> {
  const [bookData,setBookData] = useState([]);
 const classes=useStyles();

    useEffect(()=>{
        fetch('http://localhost:8000/myLibrary')
        .then(res=>{
           return res.json();
        })
        .then(data=>{console.log(data)
        setBookData(data);
        })
    },[])
  console.log(bookData);
  const navigate=useNavigate();
  return (
      
    <div className={classes.mainDiv}>
        <Banner></Banner>
    
    <div className={classes.typo} >
    <TextField variant="standard"
    InputProps={{
      startAdornment:(
        <InputAdornment position="start">
        <Search/>
        </InputAdornment>
      )
    }}
    value="Search by author and book names" fullWidth
    />
    </div>

    <Typography variant='h6' className={classes.typo}>Trending blinks</Typography>
    <Container>
    <Grid container >
     { bookData && bookData.filter((book:DataBook)=>{
       return book.isTrending===true;
     }).map((data:DataBook)=>(
       <Grid style={{padding:'10px 0 10px 15px'}} key={data.id} data-testid={`card-${data.id}`}>
       <Cards bookData={data} btnText={data.state===""?"+ Add to Library":""} bookFunction={()=>navigate(`/bookdetailviewpage/${data.id}`)}/>
         
      </Grid>
     )
     )
    }

    </Grid>
     </Container>
     <Typography variant='h6' className={classes.typo} >Just added</Typography>
     <Container>
     <Grid container >
      { bookData && bookData.filter((book:DataBook)=>{
        return book.justAdded===true;
      }).map((data:DataBook) => (
        <Grid style={{padding:'10px 0 10px 15px'}} key={data.id} data-testid={`justadded-${data.id}`}>
        <Cards bookData={data} btnText={data.state===""?"+ Add to Library":""} bookFunction={()=>navigate(`/bookdetailviewpage/${data.id}`)}/>
          
       </Grid>
      ))
      
     }
 
     </Grid>

      </Container>

      <Typography variant='h6' className={classes.typo} >Featured audio blinks</Typography>
     <Container>
     <Grid container >
      { bookData && bookData.filter((book:any)=>{
        return book.audio===true;
      }).map((data:DataBook) => (
        <Grid style={{padding:'10px 0 10px 15px'}} key={data.id} >
        <Cards bookData={data} btnText={data.state===""?"+ Add to Library":""} bookFunction={()=>navigate(`/bookdetailviewpage/${data.id}`)}/>          
       </Grid>
      ))
      
     }
 
     </Grid>

      </Container>
     </div>
  )
}


export default SeacrhBar