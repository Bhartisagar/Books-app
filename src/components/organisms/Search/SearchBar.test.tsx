import { render, screen } from "@testing-library/react";
import SeacrhBar from "./SeacrhBar";
import { MemoryRouter } from "react-router";
import renderer from 'react-test-renderer'

test("first snapshot",()=>{
    const component = renderer.create(
     <MemoryRouter><SeacrhBar/></MemoryRouter>   
    )
    let tree=component.toJSON();

    expect(tree).toMatchSnapshot();
})
test('shoult render Trending blinks heading',()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const heading1 = screen.getByRole("heading",{name:'Trending blinks'})
    expect(heading1).toBeInTheDocument();
})
test('shoult render Just added heading',()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const heading1 = screen.getByRole("heading",{name:'Just added'})
    expect(heading1).toBeInTheDocument();
})
test('shoult render Featured audio blinks heading',()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const heading1 = screen.getByRole("heading",{name:'Featured audio blinks'})
    expect(heading1).toBeInTheDocument();
})
test('should render mapp data',async()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const card = await screen.findByTestId("card-2")
    expect(card).toBeInTheDocument();
})
test('should render mapp data button',async()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const card = await screen.findByTestId("headings-3")
    expect(card).toBeInTheDocument();
})
test('should render mapp data just added book',async()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const card = await screen.findByTestId("justadded-9")
    expect(card).toBeInTheDocument();
})
test('should render mapp data buttons',async()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const card = await screen.findByTestId("button-7")
    expect(card).toBeInTheDocument();
})
test('should render mapp dat',async()=>{
    render(<MemoryRouter><SeacrhBar/></MemoryRouter>)
    const card = await screen.findByTestId("card-2")
    expect(card).toBeInTheDocument();
})
