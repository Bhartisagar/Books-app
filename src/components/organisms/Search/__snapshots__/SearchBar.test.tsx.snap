// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot 1`] = `
<div
  className="makeStyles-mainDiv-1"
>
  <div
    className="MuiBox-root MuiBox-root-8 makeStyles-banner-7"
  >
    <div
      className="makeStyles-heading-4"
    >
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
         
        Explore Books on
      </h4>
      <h4
        className="MuiTypography-root MuiTypography-h4"
      >
         
        entrepreneurship
      </h4>
      <div
        className="makeStyles-subtitlesCon-6"
      >
        <h6
          className="MuiTypography-root makeStyles-subtitles-5 MuiTypography-subtitle2"
        >
          Everything you need to know about thriving on a
        </h6>
        <h6
          className="MuiTypography-root makeStyles-subtitles-5 MuiTypography-subtitle2"
        >
          shoestring budget, making your first million, and hiring 
        </h6>
        <h6
          className="MuiTypography-root makeStyles-subtitles-5 MuiTypography-subtitle2"
        >
          right from the start.
        </h6>
      </div>
    </div>
    <img
      alt="banner"
      src="girl.svg"
    />
  </div>
  <div
    className="makeStyles-typo-2"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
        onClick={[Function]}
      >
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionStart"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value="Search by author and book names"
        />
      </div>
    </div>
  </div>
  <h6
    className="MuiTypography-root makeStyles-typo-2 MuiTypography-h6"
  >
    Trending blinks
  </h6>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    />
  </div>
  <h6
    className="MuiTypography-root makeStyles-typo-2 MuiTypography-h6"
  >
    Just added
  </h6>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    />
  </div>
  <h6
    className="MuiTypography-root makeStyles-typo-2 MuiTypography-h6"
  >
    Featured audio blinks
  </h6>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiGrid-root MuiGrid-container"
    />
  </div>
</div>
`;
